Gems/LyShine/Code/Editor/Animation/UiAnimViewDialog.cpp

```diff
@@ -1385,6 +1377,22 @@ void CUiAnimViewDialog::keyPressEvent(QKeyEvent* event)
     return QMainWindow::keyPressEvent(event);
 }
 
+//////////////////////////////////////////////////////////////////////////
+void CUiAnimViewDialog::closeEvent([[maybe_unused]] QCloseEvent* event)
+{
+    m_wndKeyPropertiesDock->hide();
+    m_wndCurveEditorDock->hide();
+}
+
+void CUiAnimViewDialog::showEvent([[maybe_unused]] QShowEvent* event)
+{
+    m_wndKeyPropertiesDock->show();
+    if (m_lastMode == ViewMode::Both)
+    {
+        m_wndCurveEditorDock->show();
```

Comment:
shouldn't this be also shown if m_lastMode is ViewMode::CurveEditor ?

Reply:
When the mode is ViewMode::CurveEditor, it does something unique by hiding the dock window and making the curve editor the central widget, so if we showed the curve editor dock here we'd see an empty window.